src/jshint.js

```diff
@@ -1424,6 +1429,12 @@ var JSHINT = (function() {
       } else if ((!left.identifier || isReserved(left)) && left.id !== "." && left.id !== "[") {
         warning("W017", this);
       }
+      if (left && left.identifier) {
```

Comment:
thanks, I knew that but hadn't thought to test it. In this case `left` will not be an identifier but a punctuator. `this` will be the punctuator `++`.
Adding tests and comments...